### PR TITLE
build: Add Dockerfile

### PR DESCRIPTION
**Pull request by parithosh**
# Describe the feature
This PR adds a dockerfile, allowing the user to build and use `reth` as a docker image. The dockerfile is accepts a `BUILD_PROFILE` as an argument, so passing `-e BUILD_PROFILE=release` or `-e BUILD_PROFILE=debug` during the `docker build` stage will allow the user to build with a different profile. The dockerfile is split into two steps, to allow for smaller release images. 

# Additional context
Docker images help users more easily try out `reth`. Its additionally easier to integrate into widespread deployment systems.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  libssl-dev \
+  ca-certificates \
+  gnupg2 \
+  software-properties-common && apt-key update \
+  && add-apt-repository -y ppa:ethereum/ethereum
+
+# Install Geth as it's a dependency for reth
+RUN apt-get update && apt-get install -y ethereum \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/*
```
Comment (gakonst) [inline Dockerfile:31]: These should not be required..geth is def not a dependency to run or build reth, we just use it in some integration tests. let's remove these. you should also be able to use a smaller image, i suspect.

https://github.com/foundry-rs/foundry/blob/master/Dockerfile#L28

**Review by gakonst — CHANGES_REQUESTED** [review]
thanks! can make the image slimmer i think

File: Dockerfile (hunk 1 above)
Comment (parithosh) [reply Dockerfile:31]: Apologies, I got the dependency part from the `reth` docs where it states: `To use Reth, you must first install Geth`. I can make a commit to revert the geth install. I can switch it to a smaller image as well. I mainly kept the `ubuntu` image as its got everything needed for debugging/testing if needed.

Comment (gakonst) [reply Dockerfile:31]: We should make the readme better then, it's only to run integration tests. Let's indeed switch to smaller image and w/o the geth stuff. People can make debug images separately, let's host a production image.

**Comment by onbjerg** [comment]
Personally I am not too keen adding a Docker image at this stage since it lead to a lot of bikeshedding in Foundry, and we are not really at a stage where we know exactly how this is going to be structured outside of a Dockerfile even. Something to keep in mind :smile:

File: Dockerfile (hunk 1 above)
Comment (parithosh) [reply Dockerfile]: I've updated the image now, it should be ~50MB at this point and there's no geth anymore.

**Review by gakonst — APPROVED** [review]
I'm good with merging this as-is @onbjerg with the expectation that we won't modify this more until we know what's up deeper.

**Comment by onbjerg** [comment]
Sounds good

**Pull request merged**